### PR TITLE
translate resnet

### PR DESCRIPTION
**Pull request by MARD1NO**
修改了Resnet 中英文章节

### CODE REVIEW

File: cn/docs/adv_examples/resnet.md [hunk 1]
```diff
@@ -56,42 +56,42 @@ ImageNet大规模视觉识别挑战赛（ILSVRC），常称为ImageNet竞赛，
...
+  - 下载我们制作的 Imagenet(2012) [迷你数据集](https://oneflow-public.oss-cn-beijing.aliyuncs.com/online_document/dataset/imagenet/mini-imagenet.zip) 解压放入data目录
+  - 或者：制作完整 OFRecord 格式的 ImageNet 数据集（见下文进阶部分）
 
-我们提供了通用脚本：train.sh和inference.sh，它们适用于此仓库下所有cnn网络模型的训练、验证、推理。您可以通过设置参数使用不同的模型、数据集来训练/推理。
+我们提供了通用脚本：`train.sh` 和 `inference.sh`，它们适用于此仓库下所有cnn网络模型的训练、验证、推理。您可以通过设置参数使用不同的模型、数据集来训练/推理。
 
  **关于模型的说明：** 
 
-> 默认情况下，我们使用resnet50，您也可以通过改动脚本中的--model参数指定其他模型，如：--model="resnet50"，--model="vgg"等。
+> 默认情况下，我们使用resnet50，您也可以通过改动脚本中的--model参数指定其他模型，如：`--model="resnet50"`，`--model="vgg"` 等。
```
Comment (doombeaker) [inline cn/docs/adv_examples/resnet.md:79]: 的--model参数 两边的空格

File: cn/docs/adv_examples/resnet.md [hunk 2]
```diff
@@ -209,26 +209,26 @@ validation: epoch 0, iter 195, top_1: 0.773237, top_k: 0.936078, samples/s: 1692
...
 
-从3轮的评估结果来看，我们的模型在Imagenet(2012)上已经达到了77.32+%的top_1精度。
+从3轮的评估结果来看，我们的模型在 ImageNet(2012) 上已经达到了77.32+%的 top1 精度。
 
-最后，恭喜你！完成了Resnet模型在ImageNet上完整的训练/验证、推理和评估，为自己鼓个掌吧！
+最后，恭喜你！完成了 Resnet 模型在 ImageNet 上完整的训练/验证、推理和评估，为自己鼓个掌吧！
 
 
 
 ## 更详细的说明 Details
```
Comment (doombeaker) [inline cn/docs/adv_examples/resnet.md:220]: 这个 Details 可以去掉

File: cn/docs/adv_examples/resnet.md [hunk 3]
```diff
@@ -342,13 +342,13 @@ onnx_model_dir = 'onnx/model'
 
 **步骤二：新建一个用于推理的 job function**
 
-然后新建一个用于推理的 job function，它只包含网络结构本身，不包含读取 OFRecord 的算子，并且直接接受 numpy 数组形式的输入。可参考 resnet\_to\_onnx.py 中的 `InferenceNet`
+然后新建一个用于推理的 job function，它只包含网络结构本身，不包含读取 OFRecord 的算子，并且直接接受 numpy 数组形式的输入。可参考 `resnet\_to\_onnx.py` 中的 `InferenceNet`。
 
-**步骤三：调用 flow.onnx.export 方法**
+**步骤三：调用 `flow.onnx.export `方法**
```
Comment (doombeaker) [inline cn/docs/adv_examples/resnet.md:347]:
```
`flow.onnx.export `
```
反引号右边内部多了个空格

File: cn/docs/adv_examples/resnet.md [hunk 4]
```diff
@@ -342,13 +342,13 @@ onnx_model_dir = 'onnx/model'
...
 
-接下来代码中会调用`oneflow_to_onnx()`方法，此方法包含了核心的模型转换方法： `flow.onnx.export()` 
+接下来代码中会调用 `oneflow_to_onnx()` 方法，此方法包含了核心的模型转换方法： `flow.onnx.export()` 
 
- **flow.onnx.export** 将从 OneFlow 网络得到 ONNX 模型，它的第一个参数是上文所说的专用于推理的 job function，第二个参数是OneFlow模型路径，第三个参数是（转换后）ONNX模型的存放路径
+ **`flow.onnx.export`** 将从 OneFlow 网络得到 ONNX 模型，它的第一个参数是上文所说的专用于推理的 job function，第二个参数是 OneFlow 模型路径，第三个参数是（转换后）ONNX 模型的存放路径
```
Comment (doombeaker) [inline cn/docs/adv_examples/resnet.md:351]:
```
**`flow.onnx.export`** 
```
强调和反引号，选一个即可

File: en/docs/adv_examples/resnet.md [hunk 5]
```diff
@@ -1,373 +1,353 @@
-## 简介 Introduction
+## Introduction
 
-### 图像分类与CNN
+### Image classification and CNN
 
- **图像分类** 是指将图像信息中所反映的不同特征，把不同类别的目标区分开来的图像处理方法，是计算机视觉中其他任务，比如目标检测、语义分割、人脸识别等高层视觉任务的基础。
+**Image classification** is an image processing method that divided different features reflected in image information into different categories of targets. It is the basis of  other tasks in computer vision, such as detection, semantic segmentation, face recognition and other high-level visual tasks.
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:5]: is an image processing method that divided different features reflected in image information into different categories of targets

->
is an image processing method that divides objects into different categories from the different features reflected in image information.

Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:5]: It is the basis of  other

other前多个空格

File: en/docs/adv_examples/resnet.md [hunk 6]
```diff
@@ -1,373 +1,353 @@
...
+## Introduction
 
-### 图像分类与CNN
+### Image classification and CNN
 
- **图像分类** 是指将图像信息中所反映的不同特征，把不同类别的目标区分开来的图像处理方法，是计算机视觉中其他任务，比如目标检测、语义分割、人脸识别等高层视觉任务的基础。
+**Image classification** is an image processing method that divided different features reflected in image information into different categories of targets. It is the basis of  other tasks in computer vision, such as detection, semantic segmentation, face recognition and other high-level visual tasks. 
 
-ImageNet大规模视觉识别挑战赛（ILSVRC），常称为ImageNet竞赛，包括图像分类、物体定位，以及物体检测等任务，是推动计算机视觉领域发展最重要的比赛之一。
+ImageNet Large-scale Visual Recognition Challenge (ILSVRC), often called ImageNet copetition, including image classification, object orientation, object detection and other tasks. It is one of the most important competition to promote the development of computer vision.
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:7]: including -> includes

File: en/docs/adv_examples/resnet.md [hunk 7]
```diff
@@ -1,373 +1,353 @@
...
 ### ResNet
 
-[ResNet](https://arxiv.org/abs/1512.03385) 是2015年ImageNet竞赛的冠军。[ResNet](https://arxiv.org/abs/1512.03385) 是2015年ImageNet竞赛的冠军。目前，ResNet相对对于传统的机器学习分类算法而言，效果已经相当的出色，之后大量的检测，分割，识别等任务也都在ResNet基础上完成。
+[ResNet](https://arxiv.org/abs/1512.03385) is the champion of 2015 competition. At present, compared with traditional machine learning classification algorithm, ResNet has achieved excellent results. After that, a large number of detection, segmentation, classification and other tasks are completed on the base of ResNet. 
 
-中，提供ResNet50 v1.5的OneFlow实现。在ImageNet-2012数据集上训练90个epoch后，准确率能达到多少？
+In [OneFlow-Benchmark](https://github.com/Oneflow-Inc/OneFlow-Benchmark) repository, we provide OneFlow implementation of ResNet50 v1.5. After 90 epochs of training on ImageNet-2012 dataset, the accuracy of evaluation can reach 77.318% (Top 1), 93.622% (Top 5). 
 
-更详细的网络参数对齐工作，见[OneFlow-Benchmark的cnns](https://github.com/Oneflow-Inc/OneFlow-Benchmark/Classification/cnns)部分
+For more detailed network parameter alignment, you can refer to [OneFlow-Benchmark's cnns](https://github.com/Oneflow-Inc/OneFlow-Benchmark/Classification/cnns) part.
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:19]: For more detailed network parameter alignment
->
For more detailed network on parameter alignment

File: en/docs/adv_examples/resnet.md [hunk 8]
```diff
@@ -1,373 +1,353 @@
...
-- 安装OneFlow。
+> Don't worry, it is easy to use OneFlow. You can start OneFlow's image recognition journey with three steps as follow. 
+>
+> - Install OneFlow，you can refer to [OneFlow project home page](https://github.com/Oneflow-Inc/oneflow) to finish installation. 
+>
+> - Clone / Download [OneFlow-Benchmark](https://github.com/Oneflow-Inc/OneFlow-Benchmark) repository. 
+>
+>   `git clone git@github.com:Oneflow-Inc/OneFlow-Benchmark.git`
+>
+>   `cd  OneFlow-Benchmark/Classification/cnns`
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:61]: 这种多行的命令，可以放在三个反引号的代码块中，而不是使用一个反引号的嵌入代码

File: en/docs/adv_examples/resnet.md [hunk 9]
```diff
@@ -1,373 +1,353 @@
...
+> > By default, we use ResNet50, you can also assign other model by setting the `--model` parameter. Such as: `--model="resnet50"`, `--model="vgg"` and so on. 
+>
+> **Description of dataset**
+>
+>
+> > 1)  To get reader quickly start, we provide synthetic virtual dataset, which refers to data is generated directly in memory as a random source of neural network. 
+> >
+> > 2) At the same time, we provide a mini-dataset. You can download and unzip it into data directory,  you can start training quickly. After getting familiar with the process, readers can refer to the making dataset part to make a complete ImageNet 2012 dataset. 
+> >
+> > 3) Using OFRecord dataset can improve the efficientcy of data loading (But this is not necessary, refer to [Data Input](../basics_topics/data_input.md), OneFlow supports loading numpy data directly).
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:82]: But this is not necessary -> But this is not required

not necessary 有然并卵的意思在里面

File: en/docs/adv_examples/resnet.md [hunk 10]
```diff
@@ -1,373 +1,353 @@
...
-- 下载[OneFlow-Benchmark](https://github.com/Oneflow-Inc/OneFlow-Benchmark)仓库。
 
-  `git clone git@github.com:Oneflow-Inc/OneFlow-Benchmark.git`
+## Quick Start
 
-- 准备数据集（可选）
-  - 下载[示例数据集](https://oneflow-public.oss-cn-beijing.aliyuncs.com/datasets/imagenet_ofrecord_example/part-00000`)
-  - 或者：制作完整OFRecord格式的ImageNet数据集（见下文进阶部分）
-  - 再或者：直接使用“合成数据”。
+So, let's start OneFlow's image classification journey !
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:88]: 感叹号前不用空格

File: en/docs/adv_examples/resnet.md [hunk 11]
```diff
@@ -1,373 +1,353 @@
...
-- 或者，下载我们已训练好的模型：[resnet_v1.5_model](https://oneflow-public.oss-cn-beijing.aliyuncs.com/model_zoo/resnet_v15_of_best_model_val_top1_77318.tgz)
+[resnet50_v1.5_model](https://oneflow-public.oss-cn-beijing.aliyuncs.com/model_zoo/resnet_v15_of_best_model_val_top1_77318.tgz ) (validation accuracy: 77.318% top1，93.622% top5 )
 
-  (validation accuracy: 77.318% top1，93.622% top5 )。
+### Predict / Inference
 
-### 预测/推理
-
-设置inference.sh中的参数：MODEL_LOAD_DIR，指定好模型路径后，直接运行：
+After downloading pretrained model, unzip it and put it into the current directory. Then execute:
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:104]: 改成
After downloading pretrained model, we need unzip it and put it into the current directory.
或者
After download pretrained model, unzip it and put it into the current directory.  （都是祈使语句）

File: en/docs/adv_examples/resnet.md [hunk 12]
```diff
@@ -1,373 +1,353 @@
...
-
-设置inference.sh中的参数：MODEL_LOAD_DIR，指定好模型路径后，直接运行：
+After downloading pretrained model, unzip it and put it into the current directory. Then execute: 
 
 ```shell
 sh inference.sh
 ```
 
-此脚本将调用模型对这张老虎图片进行分类：
+This script will call the model to classify the goldfish picture:
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:110]: 为什么中文是说图片老虎，英文是金鱼。再确认下是哪边笔误了，一起改一下。

Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:110]: This script will use the model to classify the goldfish picture

其实模型用调用不是很准确，中文也可以改。 感觉“此脚本对这张图片进行分类"就可以

File: en/docs/adv_examples/resnet.md [hunk 13]
```diff
@@ -1,373 +1,353 @@
...
-Saving model to ./output/snapshots/model_save-20200629223546/snapshot_epoch_0. validation: epoch 0, iter 100, top_1: 0.074620, top_k: 0.194120, samples/s: 2014.683
-```
-
-可以看到：
-
-- 随着训练的进行，loss不断下降，而训练的top_1/top_k准确率不断提高（其中top_k默认为top_5准确率，可自定义）。
-- 每个epoch结束时，会做另外两个工作：1）执行一次验证，并打印出验证集上的top_1/top_k准确率；2）保存模型。
-- samples/s 用来指示训练/验证的执行速度，即每秒钟能处理的图片数量。
 
+> Before executing `sh evaluate.sh`, make sure you have prepared the validation dataset of ImageNet 2012. Please refer to [OneFlow-Benchmark](https://github.com/Oneflow-Inc/OneFlow-Benchmark/blob/master/Classification/cnns) repository to learn how to make validation dataset.
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:204]: 同上，如果都是祈使语句，那么应该是
Before execute `sh evaluate.sh`

File: en/docs/adv_examples/resnet.md [hunk 14]
```diff
@@ -1,373 +1,353 @@
...
-
-可以看到：
-
-- 随着训练的进行，loss不断下降，而训练的top_1/top_k准确率不断提高（其中top_k默认为top_5准确率，可自定义）。
-- 每个epoch结束时，会做另外两个工作：1）执行一次验证，并打印出验证集上的top_1/top_k准确率；2）保存模型。
-- samples/s 用来指示训练/验证的执行速度，即每秒钟能处理的图片数量。
 
+> Before executing `sh evaluate.sh`, make sure you have prepared the validation dataset of ImageNet 2012. Please refer to [OneFlow-Benchmark](https://github.com/Oneflow-Inc/OneFlow-Benchmark/blob/master/Classification/cnns) repository to learn how to make validation dataset. 
 
+From the evaluation results of the three rounds, out model has achieved 77.32+% Top1 accuracy.
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:206]: 是不是 77.32%+ 更好?

File: en/docs/adv_examples/resnet.md [hunk 15]
```diff
@@ -1,373 +1,353 @@
...
+## Details
 
-## 更详细的说明 Details
+### Distributed training
 
-### 分布式训练
+**Simple and easy-to-use distributed training is one of OneFlow's main features**
 
-**简单而易用的分布式，是OneFlow的主打特色之一。**
+OneFlow is designed to support efficient distributed training natively. Especially for distributed data parallelism, user do not have to worry about how to divide and synchronize the data when the algorithm expands from single machine to multiple machines. That is to say, in OneFlow, User only need to write algorithm from the view of single machine, and the code automatically has the ability of distributed training.
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:218]: when the algorithm expands from single machine to multiple machines.
->
when the algorithm is extended from single machine to multiple machines.

File: en/docs/adv_examples/resnet.md [hunk 16]
```diff
@@ -1,373 +1,353 @@
...
 
+#### How to configure and run distributed training?
 
-#### 如何配置并运行分布式训练？
+We still use the code shown in the "Quick Start", in `train.sh`, the distributed configuration is easily accomplished by specifying the number of nodes (machines) with `--num_nodes`, the IP address of the nodes with `--node_ips`, and the number of devices to be used on each node with `--gpu_num_per_node`. 
 
-还是以上面"快速开始"部分演示的代码为例，在`train.sh`中，只要用`--num_nodes` 指定节点（机器）个数，同时用`--node_ips`指定节点的ip地址，然后用`--gpu_num_per_node`指定每个节点上使用的卡数，就轻松地完成了分布式的配置。
-
-例如，想要在2机8卡上进行分布式训练，像下面这样配置：
+For example, we want to do distributed training on  2 machines with 8 devices, configure it like this:
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:225]: we want to do distributed training on  2 machines with 8 devices, configure it like this:
->
if we want to run distributed training on 2 machines with 8 devices, we will configure it like this:

File: en/docs/adv_examples/resnet.md [hunk 17]
```diff
@@ -1,373 +1,353 @@
...
-
-
-### 混合精度训练与预测
+After the program starts, you can see through the command `watch -n 0.1 nvidia-smi` that both machines' devices start working. After a while, the output is printed on the screen of the first machine set by `--node_ips`. 
 
-目前，OneFlow已经原生支持半精度/全精度的混合精度训练。训练时，模型参数（权重）使用float16进行训练，同时保留float32用作梯度更新和计算过程。由于参数的存储减半，会带来训练速度的提升。训练时，模型参数（权重）使用float16进行训练，同时保留float32用作梯度更新和计算过程。由于参数的存储减半，会带来训练速度的提升。
+### Hybrid precision training and predicting
 
-在OneFlow中开启半精度/全精度的混合精度训练模式，ResNet50的训练速度理论上能达到`1.7`倍的加速。
+Currently, OneFlow supports float16/float32 hybrid precision training. During training, the model parameters are trained using float16 while retaining float32 as the gradient update and calculation process. Since the storage of parameters is halved, the training speed will be improved.
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:247]: 这个要确认下我们训练的全精度是 32位还是64位，防止笔误写错了

File: en/docs/adv_examples/resnet.md [hunk 18]
```diff
@@ -1,373 +1,353 @@
...
 
-## 进阶 Advanced
 
-### 参数对齐
+## Advanced
 
-Oneflow的ResNet50实现，为了保证和[英伟达的Mxnet版实现](https://github.com/NVIDIA/DeepLearningExamples/tree/master/MxNet/Classification/RN50v1.5)对齐，我们从learning rate学习率，优化器Optimizer的选择，数据增强的图像参数设定，到更细的每一层网络的形态，bias,weight初始化等都做了细致且几乎完全一致的对齐工作。具体的参数对齐工作，请参考：[OneFlow-Benchmark](https://github.com/Oneflow-Inc/OneFlow-Benchmark/Classification/cnns)仓库具体的参数对齐工作，请参考：[OneFlow-Benchmark](https://github.com/Oneflow-Inc/OneFlow-Benchmark/Classification/cnns)仓库
+### Parameters alignment
 
+OneFlow's ResNet50 implementation is aligned with Nvidia's Mxnet edition. We've made careful and almost identical alignment from the learning rate, optimizer, image augmentation to finer per-layer network configuration, bias, weight initialization, and more. The detailed parameters alignment please refer to [OneFlow-Benchmark](https://github.com/Oneflow-Inc/OneFlow-Benchmark/blob/master/Classification/cnns) repository.
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:268]: Nvidia's Mxnet edition -> Nvidia's Mxnet version

Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:268]: For the detailed parameters alignment,  please refer to

Comment (daquexian) [reply en/docs/adv_examples/resnet.md:268]: Mxnet 应该改成 MXNet？Nvidia 好像也是改成 NVIDIA 会更常见

File: en/docs/adv_examples/resnet.md [hunk 19]
```diff
@@ -1,373 +1,353 @@
...
 
+We provide complete code: [resnet\_to\_onnx.py](https://github.com/Oneflow-Inc/OneFlow-Benchmark/blob/master/Classification/cnns/resnet_to_onnx.py), it can help you complete the transformation and testing of the model. 
 
+ **Step1: ** Download the pretrain model: [resnet50_v1.5_model](https://oneflow-public.oss-cn-beijing.aliyuncs.com/model_zoo/resnet_v15_of_best_model_val_top1_77318.tgz ), unzip it and put it into current directory
 
-#### 将ImageNet转换成OFRecord
+ **Step2: ** Execute `python3 resnet_to_onnx.py `
 
-在OneFlow中，提供了将原始ImageNet-2012数据集文件转换成OFRecord格式的脚本。如果您已经准备好了ImageNet-2012数据集(训练集和验证集)，并且训练集/验证集的格式如下：如果您已经准备好了ImageNet-2012数据集(训练集和验证集)，并且训练集/验证集的格式如下：
+This code will complete the transformation of the OneFlow model -> ONNX model, and then use the ONNX Runtime to load the transformed model to test the individual images. The test picture is as follows:
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:302]: the OneFlow model -> ONNX model

 the OneFlow model to ONNX model

File: en/docs/adv_examples/resnet.md [hunk 20]
```diff
@@ -1,373 +1,353 @@
...
+We have introduced how to convert OneFlow's ResNet model to ONNX model and give an example of using the onnx runtime to make predictions in above example. Similarly, you can follow the steps to complete the transformation of your training ResNet model or other models. 
 
-如果您尚未下载过Imagenet数据集，请自行下载和准备以下文件：
+**Step1: Save the model's weight**
 
-- ILSVRC2012_img_train.tar
+First you should specify the OneFlow model path, and then specify the transformed ONNX model storage path, like the following example. 
 
-- ILSVRC2012_img_val.tar
+首先指定待转换的OneFlow模型路径，然后指定转换后的ONNX模型存放路径，例如示例中：
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:329]: 这是一句漏掉的中文？

File: en/docs/adv_examples/resnet.md [hunk 21]
```diff
@@ -1,373 +1,353 @@
...
+onnx_model_dir = 'onnx/model'
 ```
 
-```shell
-这一步主要是将ILSVRC2012_img_train.tar和ILSVRC2012_img_val.tar解压缩，生成train、validation文件夹。train文件夹下是1000个虚拟lebel分类文件夹(如：n01443537)，训练集图片解压后根据分类放入这些label文件夹中；validation文件夹下是解压后的原图。
-```
+**Step2: Create a new job function for inference**
 
-**步骤二：validation数据处理**
+Then, we create a new job function for inference, which only contains the network structure, except the operator to read the OFRecord, and accepts the form of numpy array input. You can refer to the `InferenceNet` in `resnet_to_onnx.py`.
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:339]: except the operator to read the OFRecord
->
does not contain the operator to read the OFRecord

File: en/docs/adv_examples/resnet.md [hunk 22]
```diff
@@ -1,373 +1,353 @@
...
+**Step3: Call `flow.onnx.export` method**
 
-```shell
-python3 preprocess_imagenet_validation_data.py  ../data/imagenet/validation
-# 参数 ../data/imagenet/validation为ILSVRC2012_img_val.tar解压后验证集图像存放的路径。
-```
+In the following code, we call the `oneflow_to_onnx()` method, this method includes the core model transformation method: `flow.onnx.export()`. 
 
-处理后项目文件夹格式如下：
+**`flow.onnx.export`** will obtain ONNX model from OneFlow network, its first parameter is the job function used to infer. The second parameter is OneFlow model path, the third parameter is the save path of ONNX model.
```
Comment (doombeaker) [inline en/docs/adv_examples/resnet.md:345]: its first parameter is the job function used to infer
->
its first parameter is the job function is used to infer

Comment (MARD1NO) [reply en/docs/adv_examples/resnet.md:345]: 存疑

**Pull request merged**